### PR TITLE
Added testing for DragAndDropHelper and a parameter to force touch behaviour

### PR DESCRIPTION
**Pull request by Andres-CT98**
Created unit test cases for the utility DragAndDropHelper.

Added an optional parameter to DragAndDropHelper to force the use of touch controls.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2,24 +2,17 @@
 # yarn lockfile v1
```
Comment (JiuqingSong) [inline yarn.lock:2]: Is there any special reason to change this file?

File: tslint.json [hunk 2]
```diff
@@ -78,5 +78,8 @@
     "rulesDirectory": [
         "node_modules/tslint-microsoft-contrib",
         "node_modules/tslint-eslint-rules/dist/rules"
-    ]
+    ],
+    "compilerOptions": {
```
Comment (JiuqingSong) [inline tslint.json:82]: why need to change this file?

File: yarn.lock (hunk 1 above)
Comment (Andres-CT98) [reply yarn.lock:2]: No, I do not know why this file changed

File: tslint.json (hunk 2 above)
Comment (Andres-CT98) [reply tslint.json:82]: It was an attempt to fix an issue with paths not being recognized for test cases. It was committed by accident

File: yarn.lock (hunk 1 above)
Comment (JiuqingSong) [reply yarn.lock:2]: please revert all unnecessary changes. All changes in a PR should be intentional.

Comment (Andres-CT98) [reply yarn.lock:2]: I know that, but I want to understand why it happened

Comment (JiuqingSong) [reply yarn.lock:2]: I have no idea why. Please just revert it.

Comment (Andres-CT98) [reply yarn.lock:2]: I tried to do it when I pushed the revert for the other file, but yarn. lock just regressed to its current state.

File: packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts [hunk 3]
```diff
@@ -72,42 +73,50 @@ export default class DragAndDropHelper<TContext, TInitValue> implements Disposab
         private context: TContext,
         private onSubmit: (context: TContext, trigger: HTMLElement) => void,
         private handler: DragAndDropHandler<TContext, TInitValue>,
-        private zoomScale: number
+        private zoomScale: number,
+        private forceMobile?: boolean /* For testing touch behaviour*/
     ) {
-        trigger.addEventListener(MOUSE_EVENT_INFO.MOUSEDOWN, this.onMouseDown);
+        if (this.forceMobile) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:79]: You can make it simpler by:
```ts
this.dndMouse = forceMobile || Browser.isMobileOrTablet ? MOUSE_EVENT_INFO_MOBILE : MOUSE_EVENT_INFO_DESKTOP

File: packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts [hunk 4]
```diff
@@ -72,42 +73,50 @@ export default class DragAndDropHelper<TContext, TInitValue> implements Disposab
         private context: TContext,
         private onSubmit: (context: TContext, trigger: HTMLElement) => void,
         private handler: DragAndDropHandler<TContext, TInitValue>,
-        private zoomScale: number
+        private zoomScale: number,
+        private forceMobile?: boolean /* For testing touch behaviour*/
```
[the remark this answers is not in the document]
Comment (Andres-CT98) [reply packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:77]: What is the correct word to use?

[the remark this answers is not in the document]
Comment (Andres-CT98) [reply packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:77]: Should I add the same for zoomScale?

[the remark this answers is not in the document]
Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:77]: no word, just remove "private".

And yes, similar with zoomScale

File: packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts [hunk 5]
```diff
@@ -67,22 +67,21 @@ export default class DragAndDropHelper<TContext, TInitValue> implements Disposab
      * so that the handler object knows which element it is triggered from.
      * @param onSubmit A callback that will be invoked when event handler in handler object returns true
      * @param handler The event handler object, see DragAndDropHandler interface for more information
+     * @param zoomScale The zoom scale of the editor
+     * @param forceMobile A boolean to force the use of touch controls for the helper
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:71]: You should still mention this is used for test code only to help others understand why we have this parameter.

Comment (Andres-CT98) [reply packages/roosterjs-editor-plugins/lib/pluginUtils/DragAndDropHelper.ts:71]: It can be used for more than testing

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**